### ISSUE TITLE
OSR_DEFAULT_AXIS_MAPPING_STRATEGY config option is disabled after creation of an OGRSpatialReference

### ISSUE THREAD

**Issue by sgillies** [OSGeo/gdal#6084]
My coworkers observed that the OSR_DEFAULT_AXIS_MAPPING_STRATEGY config option is disabled after creation of an OGRSpatialReference. Can this be changed? The project generally explains config options as feature toggles that can be turned on and off as needed and it's surprising when this is not the case.

I'm not fluent enough in C++ to understand how the strategy is frozen. Is it done here https://github.com/OSGeo/gdal/blob/master/ogr/ogrspatialreference.cpp#L189? Is GDAL_CACHEMAX frozen in the same way? Are there other config options with the same behavior?

**Comment by rouault** [comment]
The frozen characteristics comes from the "static" keyword at https://github.com/OSGeo/gdal/blob/master/ogr/ogrspatialreference.cpp#L194. Just removing it will make the GetDefaultAxisMappingStrategy() function to be called each time

**Comment by rouault** [comment]
> Is GDAL_CACHEMAX frozen in the same way? 

yes, the first time GDALGetCacheMax() is called (which is done internally by GDAL itself), or then programatically through the GDALSetCacheMax64() function

> Are there other config options with the same behavior?

yes, most certainly (none that comes immediately to mind). For some that are in hot code paths where getting the option could be too costly

**Comment by sgillies** [comment]
Thanks @rouault !
